src/server/editorServices.ts

```diff
@@ -547,9 +547,11 @@ namespace ts.server {
             }
             switch (response.kind) {
                 case ActionSet:
+                    project.resolutionCache.clear();
                     this.typingsCache.updateTypingsForProject(response.projectName, response.compilerOptions, response.typeAcquisition, response.unresolvedImports, response.typings);
                     break;
                 case ActionInvalidate:
+                    project.resolutionCache.clear();
                     this.typingsCache.deleteTypingsForProject(response.projectName);
                     break;
             }

```

Comment:
Nothing needs to happen on a `PackageInstalledResponse`? (Not specific to your code, I just see there is nothing specific for it generally).

Reply:
That response happens as a result of the codefix being invoked; as far as I know the "install types" codefix is working as expected so we don't need to clear the cache here. If that's bugged then we can make this nonconditional on the command type